[docs] Adding Calendly OAuth2 Guide (#27326)
expo/expo: docs/pages/guides/authentication.mdx

  <GridItem title="GitHub" protocol={['OAuth 2']} image={ASSETS.github} />
  <GridItem
    title="Google"

>   protocol = { ['OAuth 2', 'OpenID']}
>   href ="/guides/google-authentication/"

Reviewer: ```suggestion
    protocol={['OAuth 2', 'OpenID']}
    href="/guides/google-authentication/"
```

As per our Prettier configuration, this formatting is incorrect. Let's revert it.

  <GridItem title="Twitch" protocol={['OAuth 2']} image={ASSETS.twitch} />
  <GridItem title="Twitter" protocol={['OAuth 2']} image={ASSETS.twitter} />
  <GridItem title="Uber" protocol={['OAuth 2']} image={ASSETS.uber} />

>   <GridItem title="Calendly" protocol={['OAuth 2']} image={ASSETS.calendly} />

Reviewer: Let's not add this at the end of this Grid. On this page, we follow alphabetical order so let' stick to that. Please remove this line from here and add it before "Cognito".

</Box>
{/* End Uber */}

> {/* Start Calendly*/}

Reviewer: Same as above. On this page, we follow alphabetical order so let' stick to that. Please remove this line from here and add it before "Cognito".

<Box
  name="Calendly"
  createUrl="https://developer.calendly.com/getting-started"
  image={ASSETS.calendly}
>
| Website                   | Provider  | PKCE      | Auto Discovery |
| ------------------------- | --------- | --------- | -------------- |
| [Get Your Config](https://developer.calendly.com/getting-started) | OAuth 2.0 | Supported | Not Available  |
- `redirectUri` requires 2 slashes (`://`).
- Example redirectUri:
  - Standalone / development build: `myapp://*`
  - Web: `https://yourwebsite.com/*`
<Tabs tabs={["Auth Code"]}>

> <Tab>
> {/* prettier-ignore */}

Reviewer: ```suggestion
<Tab>

{/* prettier-ignore */}
```

Let's add the space so that Prettier does not auto-format the code snippet below.

> ```tsx

Reviewer: ````suggestion
```jsx Calendly Example
````

Add the example snippet title. Also, I don't see any TypeScript in the snippet below, so let's stick to using `.jsx`.

import * as WebBrowser from 'expo-web-browser';
import {
  makeRedirectUri,
  useAuthRequest,
  exchangeCodeAsync,
} from "expo-auth-session";
import React, { useEffect, useState } from "react";

> /* @info <strong>Web only:</strong> This method should be invoked on the page that the auth popup gets redirected to on web, it'll ensure that authentication is completed properly. On native this does nothing. */

Reviewer: ```suggestion
/* @info <strong>Web only:</strong> This method should be invoked on the page that the auth popup gets redirected to on web, it'll ensure that authentication is completed properly. On native, this does nothing. */
```

Missing comma.

WebBrowser.maybeCompleteAuthSession();
/* @end */

const discovery = {
  authorizationEndpoint: "https://auth.calendly.com/oauth/authorize",
  tokenEndpoint: "https://auth.calendly.com/oauth/token",
};

export default function App() {
  const [authTokens, setAuthTokens] = useState({access_token: "", refresh_token: ""});
  const [request, response, promptAsync] = useAuthRequest(
    {

>       clientId: /* @info Your client id, can hardcode or put it in .env file */ process.env.EXPO_PUBLIC_Client_ID /* @end */,

Reviewer: ```suggestion
      clientId: /* @info Your client id. Can be hardcoded or put it in .env file */ process.env.EXPO_PUBLIC_Client_ID /* @end */,
```

      usePKCE: true,
      redirectUri: makeRedirectUri({

>         native: /* @info Set up scheme property in your app.json or app.config.js. EX app.json {scheme: "myapp"} */ "myapp://" /* @end */, 

Reviewer: ```suggestion
        native: /* @info Set up scheme property in your app.json or app.config.js. For example, app.json {scheme: "myapp"} */ "myapp://" /* @end */, 
```

      }),
    },
    discovery
  );
  
  useEffect(() => {
    const exchange = async (exchangeTokenReq) => {
      try {
        const exchangeTokenResponse = await exchangeCodeAsync(
          {
            clientId: process.env.EXPO_PUBLIC_Client_ID,

>             code: /* @info Recieved Code from firs request*/ exchangeTokenReq /* @end */,

Reviewer: ```suggestion
            code: /* @info Received code from first request*/ exchangeTokenReq /* @end */,
```

Let's fix these typos.

            redirectUri: makeRedirectUri({
              native: "myapp://",
            }),
            extraParams: {
              code_verifier: request.codeVerifier,
            },
          },
          discovery
        );
        setAuthTokens(exchangeTokenResponse);
      } catch (error) {
        console.error("error", error);
      }
    };

    if (response) {
      if (response.error) {
        console.error(
          "Authentication error",
          response.params.error_description || "something went wrong"
        );
      }
      if (response.type === "success") {
        /* @info Exchange Received Code for Access Token */
        exchange( response.params.code);
        /* @end */
      }
    }
  }, [discovery, request, response]);

  return (
  <SafeAreaView>
      <View>
        <Text>0Auth2</Text>
        <Button

>           title="Connect to calendly"

Reviewer: ```suggestion
          title="Connect to Calendly"
```

Branding title.
